src/js/view/controls/components/settings/menu.js

```diff
@@ -19,22 +19,51 @@ export function SettingsMenu(onVisibility, onSubmenuAdded, onMenuEmpty) {
 
     const settingsMenuElement = createElement(SettingsMenuTemplate());
 
-    const closeOnEnter = function(evt) {
-        if (evt && evt.keyCode === 27) {
-            instance.close(evt);
+    const handleKeyDown = function(evt) {
+        if (evt) {
+            switch (evt.keyCode) {
+                case 27: // esc
+                    instance.close();
+                    break;
+                case 37: // left-arrow
+                    if (evt.target.previousElementSibling) {
+                        evt.target.previousElementSibling.focus();
+                    } else {
+                        instance.close();
+                    }
+                    break;
+                case 38: // up-arrow
+                    instance.activateSubmenu(evt.target.getAttribute('name'), true);
+                    break;
+                case 39: // right-arrow
+                    if (evt.target !== closeButton.element()) {
+                        evt.target.nextElementSibling.focus();
```

Comment:
You should check for `nextElementSibling` like you do for `previousElementSibling` as well as the closeButton. Also what about `instance.close()`? It's used in left-arrow, but not on right arrow.

Reply:
@robwalch Theres a separate keydown listener on the close button, thats created from the button function  and I account for the right arrow there:

```javascript
const closeOnButton = function(evt) {
        // Close settings menu when enter is pressed on the close button
        // or when tab or right arrow key is pressed since it is the last element in topbar
        if (evt.keyCode === 13 || evt.keyCode === 39 || (evt.keyCode === 9 && !evt.shiftKey)) {
            instance.close(evt);
        }
    };
    closeButton.show();
    closeButton.element().addEventListener('keydown', closeOnButton);
```